### DISCUSSION TITLE
Set Default Window Width

### DISCUSSION THREAD

**Discussion by ghost** [willfaught/paige#86] [Q&A] [answered]
The width of the window depends on the content. If there's less text, the window is narrow, and if there's more, it is wide.

It looks like the maximum window width is 80% - 85% of the screen, which I prefer.

How can set this max width as the default ?

**Comment by willfaught** [comment]
I’m not sure what you’re asking for. By window width, do you mean the width of the page content, e.g. the width of a blog post’s body?

If so, are you asking for that width to always be the maximum width? Even for one-word bodies? In that case, the one word would appear to the left of the screen, not centered. I can’t test this for several days, but you should be able to style #paige-content to control the width. See the narrow look section in the readme for an example. In this case, you would set its width style to 100%.

**Reply by ghost** [reply]
Yes I'm refering to the width of a blog post's body. The one word appearing on the left is expected as the content is left aligned. 

In the narrow look section I dont see a width style. 


```
[params.paige]
style = """
#paige-content > * { margin-left: auto; margin-right: auto; max-width: 40rem; }
"""
```

I tried setting `max-width` from 40rem to 100% but it doesn't work.

**Reply by willfaught** [reply]
Instead of setting max-width like that, try setting `width: 100%`:

```toml
[params.paige]
style = """
#paige-content > * { margin-left: auto; margin-right: auto; width: 100%; }
"""
```

**Reply by willfaught** [reply]
Note that with the default style, the content is still justified left; it just centers the container of the page content, so if the width of the longest line isn’t >= the max page content width, the left and right margins around the page content will be wider.

**Reply by ghost** [reply]
I dont any changes being reflected after adding:

```
[params.paige]
style = """
#paige-content > * { margin-left: auto; margin-right: auto; width: 100%; }
"""
```

**Reply by willfaught** [reply]
Maybe try min-width: 100%? I won’t have my computer for a few days, so can’t test until then.

**Reply by ghost** [reply]
Doesn't seem to work.
I'll wait till you're available.

**Reply by willfaught** [reply]
@saurabh-sm Sorry for the delay. I'd like to better understand your position. Why do you want pages to always have the same width? If there are only 3 words in the page content, why do you want them to appear to the far left of the screen?

**Reply by ghost** [reply]
I prefer all pages to have the same width for consistency. I have a few pages where the content is not wide enough, but other pages have enough content to span the maximum width that Paige allows. Each page having the same width would be consistent. 

There's a `title` and `description` in the front matter. The description I fill in is usually one - two sentences (about 15 words). 

When the actual content of the page doesn't match the front matter's `description`, it looks odd.

So in terms of the page width being wide enough, the content is already there in the `description`. The actual page content is never as low as 3 words. 

Furthermore, given that there's already a narrow look, a wide look would be nice.

**Reply by willfaught** [reply]
>When the actual content of the page doesn't match the front matter's description, it looks odd.

What do you mean by "doesn't match"? Do you mean when the content is narrower than the description?

**Reply by willfaught** [reply]
Can you paste a screenshot here?

**Reply by ghost** [reply]
> What do you mean by "doesn't match"? Do you mean when the content is narrower than the description?

Yes. Here's the screenshot:

<img width="1296" alt="Screenshot 2024-08-27 at 9 47 28 AM" src="https://github.com/user-attachments/assets/7fe401ea-a737-4e7d-aa76-9c928e0441b0">

I think headings "Foo" and "Bar" should be left aligned with "Random".

**Reply by willfaught** [reply]
@allocgator I added config parameter `paige.pages.max_width` in v0.100.0, released just now, which should do what you want. Let me know what you think.

**Reply by ghost** [reply]
```
[paige.pages]
...
...
max_width = true
``` 

doesn't seem to work :(

**Reply by willfaught** [reply]
When I apply this diff to the Paige repo:

```diff
diff --git a/exampleSite/content/content/math-typesetting.md b/exampleSite/content/content/math-typesetting.md
index e7c0b9a5..dc5bed22 100644
--- a/exampleSite/content/content/math-typesetting.md
+++ b/exampleSite/content/content/math-typesetting.md
@@ -6,6 +6,8 @@ tags = ["katex", "math", "typesetting"]
 title = "Math Typesetting"
 [paige]
 enable_math = true
+[paige.pages]
+max_width = true
 +++
 
 <!--
```

I see the expected max width applied:

<img width="1395" alt="Screenshot 2025-02-23 at 11 37 41 AM" src="https://github.com/user-attachments/assets/3af1a48e-7dbc-439e-acc8-609dbe9db6c0" />

What are you seeing?

**Reply by ghost** [reply]
I set the value in `hugo.toml`. 

I'm seeing that the width for pages depends on the text contained in it, as a result of which the width is different for different pages.

**Reply by willfaught** [reply]
I still see expected behavior:

<img width="1395" alt="Screenshot 2025-02-23 at 11 53 16 AM" src="https://github.com/user-attachments/assets/785fca0d-e955-4f9e-95e0-c45d7d3f2911" />
<img width="1395" alt="Screenshot 2025-02-23 at 11 53 52 AM" src="https://github.com/user-attachments/assets/4f02902c-42b7-45ca-88bf-bfd8c618a65b" />

What are you seeing specifically? What code are you using?

**Reply by willfaught** [reply, marked as answer]
Achieved with:

```diff
diff --git a/exampleSite/hugo.toml b/exampleSite/hugo.toml
index 0eb79cf8..acdeb222 100644
--- a/exampleSite/hugo.toml
+++ b/exampleSite/hugo.toml
@@ -49,20 +49,21 @@ web_master = "example@example.com (John Doe)"
 [params.paige.pages]
 disable_authors = true
 disable_date = true
 disable_keywords = true
 disable_next = true
 disable_prev = true
 disable_reading_time = true
 disable_series = true
 disable_toc = true
 disable_word_count = true
+max_width = true
 
 [paige.pages.base_schema]
 isAccessibleForFree = true
 isFamilyFriendly = true
 
 [params.paige.site]
 disable_breadcrumbs = true
 disable_credit = true
 
 [params.paige.subpages]
```

**Reply by ghost** [reply]
On Paige version `0.110.0`, setting `full_width = true` under `[params.paige.pages]` has the desired effect.

It looks like the documentation about setting `full_width = true` under `[paige.pages]` has no effect.

Is there a difference between `[paige.pages]` and `[params.paige.pages]`? The following content overlaps in the [parameters section](https://github.com/willfaught/paige?tab=readme-ov-file#parameters) and the [minimal look section](https://github.com/willfaught/paige?tab=readme-ov-file#minimal-look):

```
disable_authors
disable_date
disable_keywords
disable_next
disable_prev
disable_reading_time
disable_series
disable_toc
disable_word_count
```

**Reply by willfaught** [reply]
The Parameters section says:

>The parameter object is accessed in site parameters as [params.paige], and in page parameters as [paige].

The Minimal Look section says:

>If you want a more minimal look, try the following in yoursite/hugo.toml:

That's meant to say that the exact syntax in yoursite/hugo.toml should be what's shown, because it needs to be under the [params] field. If the instructions were to put it in a page's front matter, the "params." prefix would be omitted.

**Reply by willfaught** [reply]
Thought about it more, and decided to simplify this by not specifying the place to put the config and removing the `params.` prefix. Hopefully that will be clearer.

**Reply by ghost** [reply]
That will definitely be clearer.

**Comment by ghost** [comment]
I have a lot of content which worked with version 0.93.

I’ll be making a lot of changes today so that things work with 0.100.2 as they did with 0.93. 

I’ll get back soon on this. 
--
Secured with Tuta Mail:
https://tuta.com/free-email


Feb 23, 2025 at 11:57 by ***@***.***:
